LoRaEngine/LoraKeysManagerFacade/CreateEdgeDevice.cs

```diff
@@ -146,6 +148,40 @@ public CreateEdgeDevice(RegistryManager registryManager)
             return PrepareResponse(HttpStatusCode.OK);
         }
 
+        private static string ReplaceJsonWithCorrectValues(string region, string resetPin, string json, string spiSpeed, string spiDev)
+        {
+            json = json.Replace("[$region]", region);
+            json = json.Replace("[$reset_pin]", resetPin);
+
+            if (string.Equals(spiSpeed, "8", StringComparison.CurrentCultureIgnoreCase) ||
+                string.IsNullOrEmpty(spiSpeed))
+            {
+                // default case
+                json = json.Replace("[$spi_speed]", string.Empty);
+            }
+            else
+            {
+                json = json.Replace(
+                    "[$spi_speed]",
+                    ",'SPI_SPEED':{'value':'2'}");

```

Comment:
This seems weird. Why would this be a constant 2? Looks like the passed in spiSpeed is never used and should be used here?